api-docs/client-files.md

```diff
@@ -0,0 +1,318 @@
+  },
+  "fileName": "fileKey",
+  "fileSize": "10mb",
+  "fileExtension": "pdf",
+  "fileDesc": "invoice"
+}
+```
+
+#### Error
+
+If there is no client with the provided id, you will get a 404 HTTP response, with the following error
+
+```json
+{
+  "errors": ["Could not find client with id 2"]
+}
+```
+
+If there is no file with the provided id, you will get a 404 HTTP response, with the following error
+
+```json
+{
+  "errors": ["Could not find fileKey"]
+}
+```
+
+---
+
+## Get all client files
+
+```http
+  GET /api/clients/:clientId/files
+```
+
+### Response
+
+#### Success
+
+```json
+{
+  "files": [
+    {
+      "clientId": 1,
+      "interactionId": 1,
+      "createdBy": {
+        "firstName": "Sean",
+        "lastName": "White",
+        "fullName": "Sean White",
+        "timestamp": "2019-11-06T06:00:00.000Z"
+      },
+      "fileName": "fileKey",
+      "fileSize": "10mb",
+      "fileExtension": "pdf",
+      "fileDesc": "invoice"
+    }
+  ]
+}
+```
+
+#### Error
+
+If there is no client with the provided id, you will get a 404 HTTP response, with the following error
+
+```json
+{
+  "errors": ["Could not find client with id 2"]
+}
+```
+
+---
+
+## Create a client file
+
+```http
+  POST /api/clients/:clientId/file
+```
+
+### Request
+
+```json
+{
+  "clientId": 1,
+  "interactionId": 1,
+  "fileName": "fileKey",
+  "fileSize": "10mb",
+  "fileExtension": "pdf",
+  "fileDesc": "invoice"
+}
+```
+
+**NOTES**
+
+The `filename` will be included in the signed url. Not in the uploaded file.
+
+### Response
+
+#### Success
+
+```json
+{
+  "clientId": 1,
+  "interactionId": 1,
+  "createdBy": {
+    "firstName": "Sean",
+    "lastName": "White",
+    "fullName": "Sean White",
+    "timestamp": "2019-11-06T06:00:00.000Z"
+  },
+  "fileName": "fileKey",
+  "fileSize": "10mb",
+  "fileExtension": "pdf",
+  "fileDesc": "invoice"
+}
+```
+
+#### Error
+
+Validation errors will respond with HTTP status 400.
+
+```json
+{
+  "errors": ["You must provide a id"]
+}
+```
+
+---
+
+## Create multiple client files
+
+```http
+  POST /api/clients/:clientId/files
+```
+
+```json
+{
+  "files": [
+    {
+      "clientId": 1,
+      "interactionId": 1,
+      "fileName": "fileKey",
+      "fileSize": "10mb",
+      "fileExtension": "pdf",
+      "fileDesc": "invoice"
+    }
+  ]
+}
+```
+
+### Response
+
+#### Success
+
+```json
+{
+  "files": [
+    {
+      "clientId": 1,
+      "interactionId": 1,
+      "createdBy": {
+        "firstName": "Sean",
+        "lastName": "White",
+        "fullName": "Sean White",
+        "timestamp": "2019-11-06T06:00:00.000Z"
+      },
+      "fileName": "fileKey",
+      "fileSize": "10mb",
+      "fileExtension": "pdf",
+      "fileDesc": "invoice"
+    }
+  ]
+}
+```
+
+#### Error
+
+Validation errors will respond with HTTP status 400.
+
+```json
+{
+  "errors": ["You must provide a id"]
+}
+```
+
+---
+
+## Soft delete a client file
+
+```http
+DELETE /api/clients/:clientId/file/:fileKey
```

Comment:
`fileId`, not `fileKey`. We want to allow for two files with the same name, which is only possible if we use the mysql id as the unique thing instead of the fileKey name